cinder/volume/manager.py

```diff
@@ -356,18 +370,29 @@ def _count_allocated_capacity(self, ctxt: context.RequestContext,
                 pool = (self.driver.configuration.safe_get(
                     'volume_backend_name') or volume_utils.extract_host(
                     volume['host'], 'pool', True))
-        try:
-            pool_stat = self.stats['pools'][pool]
-        except KeyError:
-            # First volume in the pool
-            self.stats['pools'][pool] = dict(
-                allocated_capacity_gb=0)
-            pool_stat = self.stats['pools'][pool]
-        pool_sum = pool_stat['allocated_capacity_gb']
-        pool_sum += volume['size']
 
-        self.stats['pools'][pool]['allocated_capacity_gb'] = pool_sum
-        self.stats['allocated_capacity_gb'] += volume['size']
+        self._update_pool_allocated_capacity(pool, volume['size'])
+
+    def _update_snapshot_allocated_capacity(self, ctxt: context.RequestContext,
+                                            snapshot: objects.Snapshot,
+                                            host: str) -> None:
```

Comment:
`host` is not always `str`, it is passed as `None`, too, so this would need to be `Optional[str]`, right?

Reply:
fixed